File: plugins/capawesome-capacitor-wifi/android/src/main/java/io/capawesome/capacitorjs/plugins/wifi/classes/options/ConnectOptions.java
package io.capawesome.capacitorjs.plugins.wifi.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ConnectOptions {

    @NonNull
    private String ssid;

    @Nullable
    private String password;

    private boolean isHiddenSsid;

    public ConnectOptions(@NonNull String ssid, @Nullable String password, boolean isHiddenSsid) {
        this.ssid = ssid;
        this.password = password;
        this.isHiddenSsid = isHiddenSsid;
    }

    @NonNull
    public String getSsid() {
        return ssid;
    }

    @Nullable
    public String getPassword() {
        return password;
    }

    public boolean isHiddenSsid() {
        return isHiddenSsid;
    }
}

File: plugins/capawesome-capacitor-wifi/ios/Plugin/Classes/Options/ConnectOptions.swift
import Foundation
import Capacitor

@objc public class ConnectOptions: NSObject {
    private var ssid: String
    private var password: String?

    init(ssid: String, password: String?) {
        self.ssid = ssid
        self.password = password
    }

    func getSsid() -> String {
        return ssid
    }

    func getPassword() -> String? {
        return password
    }
}
